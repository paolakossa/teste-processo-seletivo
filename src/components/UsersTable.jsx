import React from 'react';
import { Table, Avatar, Space } from 'antd';
import styled from 'styled-components';

const TableContainer = styled.div`
  margin-top: 1rem;
  overflow-x: auto;
`;

export const StyledTable = styled(Table)`
  width: 100%;

  .ant-table {
    width: 100%;
    min-width: 600px;
  }

  .ant-table-thead > tr > th {
    border: none !important;
    border-bottom: 1px solid #d9d9d9 !important;
  }

  .ant-table-thead > tr > th::before,
  .ant-table-thead > tr > th::after {
    display: none !important;
    content: none !important;
  }

  .ant-table-column-sorters {
    border: none !important;
    background: transparent !important;
    box-shadow: none !important;
  }

  .ant-table-thead > tr > th.gray-bg {
    background-color: #f0f0f0 !important;
  }

  .ant-table-thead > tr > th.gray-bg:hover,
  .ant-table-thead > tr > th.gray-bg.ant-table-column-sort {
    background-color: #f0f0f0 !important;
  }

  @media (max-width: 768px) {
    .ant-table-thead > tr > th {
      font-size: 14px;
      padding: 8px;
    }
  }
`;

export const UsersTable = ({ users }) => {
  const columns = [
    {
      title: 'ID',
      dataIndex: 'id',
      key: 'id',
      width: '5%',
    },
    {
      title: 'Nome completo',
      key: 'name',
      sorter: (a, b) => {
        const nameA = (a.first_name + ' ' + a.last_name).toLowerCase();
        const nameB = (b.first_name + ' ' + b.last_name).toLowerCase();
        return nameA.localeCompare(nameB);
      },
      className: 'gray-bg',
      render: (record) => (
        <Space>
          <Avatar src={record.avatar} />
          <span>
            {record.first_name} {record.last_name}
          </span>
        </Space>
      ),
      width: '30%',
    },
    {
      title: 'E-mail',
      dataIndex: 'email',
      key: 'email',
      width: '25%',
      sorter: (a, b) => a.email.localeCompare(b.email),
      className: 'gray-bg',
    },
    {
      title: 'Ações',
      key: 'acoes',
      render: () => <a href="#">Mais detalhes</a>,
      width: '20%',
    },
  ];

  return (
    <TableContainer>
      <StyledTable dataSource={users} columns={columns} rowKey="id" pagination={false} />
    </TableContainer>
  );
};
