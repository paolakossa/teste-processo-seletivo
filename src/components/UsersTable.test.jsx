import React from 'react';
import { render, screen } from '@testing-library/react';
import { expect, describe, it } from 'vitest';
import { MemoryRouter } from 'react-router-dom';
import { UsersTable } from './UsersTable';

const mockUsers = [
  {
    id: 1,
    first_name: 'Maria',
    last_name: 'Silva',
    email: 'maria@example.com',
    avatar: 'https://via.placeholder.com/40',
  },
  {
    id: 2,
    first_name: 'João',
    last_name: 'Oliveira',
    email: 'joao@example.com',
    avatar: 'https://via.placeholder.com/40',
  },
];

const renderWithRouter = (ui) => {
  return render(<MemoryRouter>{ui}</MemoryRouter>);
};

describe('UsersTable Component', () => {
  it('deve renderizar corretamente com dados', () => {
    renderWithRouter(<UsersTable users={mockUsers} />);
    expect(screen.getByText('ID')).toBeInTheDocument();
    expect(screen.getByText('Nome completo')).toBeInTheDocument();
    expect(screen.getByText('E-mail')).toBeInTheDocument();
    expect(screen.getByText('Ações')).toBeInTheDocument();

    expect(screen.getByText('Maria Silva')).toBeInTheDocument();
    expect(screen.getByText('João Oliveira')).toBeInTheDocument();
  });

  it('deve corresponder ao snapshot', () => {
    const { container } = renderWithRouter(<UsersTable users={mockUsers} />);
    expect(container).toMatchSnapshot();
  });

  it('deve renderizar "Mais detalhes" para cada usuário', () => {
    renderWithRouter(<UsersTable users={mockUsers} />);
    const links = screen.getAllByText('Mais detalhes');
    expect(links.length).toBe(mockUsers.length);
  });
});
