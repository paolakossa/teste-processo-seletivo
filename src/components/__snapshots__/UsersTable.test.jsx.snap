// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UsersTable Component > deve corresponder ao snapshot 1`] = `
.c0 {
  margin-top: 1rem;
  overflow-x: auto;
}

.c1 {
  width: 100%;
}

.c1 .ant-table {
  width: 100%;
  min-width: 600px;
}

.c1 .ant-table-thead>tr>th {
  border: none!important;
  border-bottom: 1px solid #d9d9d9!important;
}

.c1 .ant-table-thead>tr>th::before,
.c1 .ant-table-thead>tr>th::after {
  display: none!important;
  content: none!important;
}

.c1 .ant-table-column-sorters {
  border: none!important;
  background: transparent!important;
  box-shadow: none!important;
}

.c1 .ant-table-thead>tr>th.gray-bg {
  background-color: #f0f0f0!important;
}

.c1 .ant-table-thead>tr>th.gray-bg:hover,
.c1 .ant-table-thead>tr>th.gray-bg.ant-table-column-sort {
  background-color: #f0f0f0!important;
}

@media (max-width: 768px) {
  .c1 .ant-table-thead>tr>th {
    font-size: 14px;
    padding: 8px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="ant-table-wrapper c1 css-dev-only-do-not-override-tjsggz"
    >
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-tjsggz"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table css-dev-only-do-not-override-tjsggz"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
              >
                <table
                  style="table-layout: auto;"
                >
                  <colgroup>
                    <col
                      style="width: 5%;"
                    />
                    <col
                      style="width: 30%;"
                    />
                    <col
                      style="width: 25%;"
                    />
                    <col
                      style="width: 20%;"
                    />
                  </colgroup>
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        ID
                      </th>
                      <th
                        aria-label="Nome completo"
                        class="ant-table-cell gray-bg ant-table-column-has-sorters"
                        scope="col"
                        tabindex="0"
                      >
                        <div
                          aria-describedby="test-id"
                          class="ant-table-column-sorters"
                        >
                          <span
                            class="ant-table-column-title"
                          >
                            Nome completo
                          </span>
                          <span
                            class="ant-table-column-sorter ant-table-column-sorter-full"
                          >
                            <span
                              aria-hidden="true"
                              class="ant-table-column-sorter-inner"
                            >
                              <span
                                aria-label="caret-up"
                                class="anticon anticon-caret-up ant-table-column-sorter-up"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="caret-up"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                  />
                                </svg>
                              </span>
                              <span
                                aria-label="caret-down"
                                class="anticon anticon-caret-down ant-table-column-sorter-down"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="caret-down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </span>
                        </div>
                      </th>
                      <th
                        aria-label="E-mail"
                        class="ant-table-cell gray-bg ant-table-column-has-sorters"
                        scope="col"
                        tabindex="0"
                      >
                        <div
                          aria-describedby="test-id"
                          class="ant-table-column-sorters"
                        >
                          <span
                            class="ant-table-column-title"
                          >
                            E-mail
                          </span>
                          <span
                            class="ant-table-column-sorter ant-table-column-sorter-full"
                          >
                            <span
                              aria-hidden="true"
                              class="ant-table-column-sorter-inner"
                            >
                              <span
                                aria-label="caret-up"
                                class="anticon anticon-caret-up ant-table-column-sorter-up"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="caret-up"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                  />
                                </svg>
                              </span>
                              <span
                                aria-label="caret-down"
                                class="anticon anticon-caret-down ant-table-column-sorter-down"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="caret-down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </span>
                        </div>
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Ações
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="1"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        1
                      </td>
                      <td
                        class="ant-table-cell gray-bg"
                      >
                        <div
                          class="ant-space css-dev-only-do-not-override-tjsggz ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                        >
                          <div
                            class="ant-space-item"
                          >
                            <span
                              class="ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-tjsggz"
                            >
                              <img
                                src="https://via.placeholder.com/40"
                              />
                            </span>
                          </div>
                          <div
                            class="ant-space-item"
                          >
                            <span>
                              Maria
                               
                              Silva
                            </span>
                          </div>
                        </div>
                      </td>
                      <td
                        class="ant-table-cell gray-bg"
                      >
                        maria@example.com
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a
                          href="#"
                        >
                          Mais detalhes
                        </a>
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="2"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        2
                      </td>
                      <td
                        class="ant-table-cell gray-bg"
                      >
                        <div
                          class="ant-space css-dev-only-do-not-override-tjsggz ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                        >
                          <div
                            class="ant-space-item"
                          >
                            <span
                              class="ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-tjsggz"
                            >
                              <img
                                src="https://via.placeholder.com/40"
                              />
                            </span>
                          </div>
                          <div
                            class="ant-space-item"
                          >
                            <span>
                              João
                               
                              Oliveira
                            </span>
                          </div>
                        </div>
                      </td>
                      <td
                        class="ant-table-cell gray-bg"
                      >
                        joao@example.com
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <a
                          href="#"
                        >
                          Mais detalhes
                        </a>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
